src/d3d8/d3d8_device.cpp

```diff
@@ -970,6 +970,7 @@ namespace dxvk {
 
     D3D8StateBlock* block = reinterpret_cast<D3D8StateBlock*>(Token);
     m_stateBlocks.erase(block);
+    delete block;
```

Comment:
Wouldn't it be more elegant to turn m_stateBlocks into a vector of unique_ptrs?

Reply:
@K0bin I will PR something that reworks this bit post-upstreaming. In short, native d3d8 uses incremental token values to identify state blocks, not their addresses, so m_stateBlocks needs to turn into an unordered_map of DWORD and D3D8StateBlock*. The token DWORD will start at 0 and monotonically increase throughout the lifecycle of a device, whenever a new state block is created.